### ISSUE TITLE
OOM when eval a relative large package

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#184]
_Originally opened by @adieu in https://github.com/cuelang/cue/issues/184_

I've been playing with cue lately with the ansible module schema definitions.

There are 3458 yaml files. After import, there are 3003 cue files. (There are some errors during import, I'm still trying to figure them out)

When I tried to do an eval to get the snapshot with `cue eval ./... > snapshot`, I got an OOM error.

```
fatal error: runtime: out of memory

runtime stack:
runtime.throw(0xc1acec, 0x16)
	/usr/local/go/src/runtime/panic.go:774 +0x72
runtime.sysMap(0xc09c000000, 0x4000000, 0x12135d8)
	/usr/local/go/src/runtime/mem_linux.go:169 +0xc5
runtime.(*mheap).sysAlloc(0x11fae40, 0x2000, 0x2000, 0xc000067ed8)
	/usr/local/go/src/runtime/malloc.go:701 +0x1cd
runtime.(*mheap).grow(0x11fae40, 0x1, 0xffffffff)
	/usr/local/go/src/runtime/mheap.go:1255 +0xa3
runtime.(*mheap).allocSpanLocked(0x11fae40, 0x1, 0x12135e8, 0x7edae50c31c8)
	/usr/local/go/src/runtime/mheap.go:1170 +0x266
runtime.(*mheap).alloc_m(0x11fae40, 0x1, 0x450010, 0x7edae50c31c8)
	/usr/local/go/src/runtime/mheap.go:1022 +0xc2
runtime.(*mheap).alloc.func1()
	/usr/local/go/src/runtime/mheap.go:1093 +0x4c
runtime.systemstack(0x0)
	/usr/local/go/src/runtime/asm_amd64.s:370 +0x66
runtime.mstart()
	/usr/local/go/src/runtime/proc.go:1146

goroutine 1 [running]:
runtime.systemstack_switch()
	/usr/local/go/src/runtime/asm_amd64.s:330 fp=0xc0089f25c8 sp=0xc0089f25c0 pc=0x4584f0
runtime.(*mheap).alloc(0x11fae40, 0x1, 0x7edae5010010, 0x0)
	/usr/local/go/src/runtime/mheap.go:1092 +0x8a fp=0xc0089f2618 sp=0xc0089f25c8 pc=0x4258fa
runtime.(*mcentral).grow(0x11fb5f8, 0x0)
	/usr/local/go/src/runtime/mcentral.go:255 +0x7b fp=0xc0089f2658 sp=0xc0089f2618 pc=0x41784b
runtime.(*mcentral).cacheSpan(0x11fb5f8, 0x203026)
	/usr/local/go/src/runtime/mcentral.go:106 +0x2fe fp=0xc0089f26b8 sp=0xc0089f2658 pc=0x41736e
runtime.(*mcache).refill(0x7edaf0188d98, 0x10)
	/usr/local/go/src/runtime/mcache.go:138 +0x85 fp=0xc0089f26d8 sp=0xc0089f26b8 pc=0x416e15
runtime.(*mcache).nextFree(0x7edaf0188d98, 0x10, 0x38, 0xbbd1e0, 0xc0088018c0)
	/usr/local/go/src/runtime/malloc.go:854 +0x87 fp=0xc0089f2710 sp=0xc0089f26d8 pc=0x40b8e7
runtime.mallocgc(0x70, 0xbdf640, 0xc004b59a01, 0xc09bff1c40)
	/usr/local/go/src/runtime/malloc.go:1022 +0x793 fp=0xc0089f27b0 sp=0xc0089f2710 pc=0x40c223
runtime.newobject(0xbdf640, 0x1)
	/usr/local/go/src/runtime/malloc.go:1151 +0x38 fp=0xc0089f27e0 sp=0xc0089f27b0 pc=0x40c618
cuelang.org/go/cue.rewriteCopy(0xc0088018c0, 0xd2e0c0, 0xc09b9b71f0, 0x0, 0x0, 0xc0089f2c00)
	/Users/mpvl/dev/release/cue/cue/copy.go:34 +0x5ef fp=0xc0089f2a58 sp=0xc0089f27e0 pc=0x74647f
cuelang.org/go/cue.rewrite(0xc0088018c0, 0xd2e0c0, 0xc09b9b71f0, 0xc461d8, 0xd2dee0, 0x2)
	/Users/mpvl/dev/release/cue/cue/rewrite.go:26 +0x48 fp=0xc0089f2a98 sp=0xc0089f2a58 pc=0x7733a8
cuelang.org/go/cue.(*list).rewrite(0xc09b9af880, 0xc0088018c0, 0xc461d8, 0xd2dd00, 0xc09b9af880)
	/Users/mpvl/dev/release/cue/cue/rewrite.go:130 +0x53 fp=0xc0089f2b00 sp=0xc0089f2a98 pc=0x774623
cuelang.org/go/cue.rewrite(0xc0088018c0, 0xd2dd00, 0xc09b9af880, 0xc461d8, 0x0, 0xc004b59b00)
	/Users/mpvl/dev/release/cue/cue/rewrite.go:30 +0x77 fp=0xc0089f2b40 sp=0xc0089f2b00 pc=0x7733d7
cuelang.org/go/cue.(*context).copy(...)
	/Users/mpvl/dev/release/cue/cue/copy.go:19
cuelang.org/go/cue.rewriteCopy(0xc0088018c0, 0xd2e0c0, 0xc09b9b7180, 0x0, 0x0, 0xc01336da00)
	/Users/mpvl/dev/release/cue/cue/copy.go:52 +0x8fe fp=0xc0089f2db8 sp=0xc0089f2b40 pc=0x74678e
cuelang.org/go/cue.rewrite(0xc0088018c0, 0xd2e0c0, 0xc09b9b7180, 0xc461d8, 0xd2e0c0, 0xc09bffbe30)
	/Users/mpvl/dev/release/cue/cue/rewrite.go:26 +0x48 fp=0xc0089f2df8 sp=0xc0089f2db8 pc=0x7733a8
cuelang.org/go/cue.(*context).copy(...)
	/Users/mpvl/dev/release/cue/cue/copy.go:19
cuelang.org/go/cue.rewriteCopy(0xc0088018c0, 0xd2e0c0, 0xc09b9b6fc0, 0x0, 0x0, 0xc01336da00)
	/Users/mpvl/dev/release/cue/cue/copy.go:52 +0x8fe fp=0xc0089f3070 sp=0xc0089f2df8 pc=0x74678e
cuelang.org/go/cue.rewrite(0xc0088018c0, 0xd2e0c0, 0xc09b9b6fc0, 0xc461d8, 0xd2dd00, 0xc09bff1b80)
	/Users/mpvl/dev/release/cue/cue/rewrite.go:26 +0x48 fp=0xc0089f30b0 sp=0xc0089f3070 pc=0x7733a8
cuelang.org/go/cue.(*context).copy(...)
	/Users/mpvl/dev/release/cue/cue/copy.go:19
cuelang.org/go/cue.rewriteCopy(0xc0088018c0, 0xd2e0c0, 0xc09b9b6e70, 0x0, 0x0, 0x1211700)
	/Users/mpvl/dev/release/cue/cue/copy.go:52 +0x8fe fp=0xc0089f3328 sp=0xc0089f30b0 pc=0x74678e
cuelang.org/go/cue.rewrite(0xc0088018c0, 0xd2e0c0, 0xc09b9b6e70, 0xc461d8, 0xd2e0c0, 0xc09bffb3b0)
	/Users/mpvl/dev/release/cue/cue/rewrite.go:26 +0x48 fp=0xc0089f3368 sp=0xc0089f3328 pc=0x7733a8
cuelang.org/go/cue.(*context).copy(...)
	/Users/mpvl/dev/release/cue/cue/copy.go:19
cuelang.org/go/cue.(*structLit).binOp(0xc09b608460, 0xc0088018c0, 0xd26ac0, 0xc004993780, 0xc004940001, 0xd30b60, 0xc004945260, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/binop.go:672 +0x164a fp=0xc0089f3788 sp=0xc0089f3368 pc=0x7335aa
cuelang.org/go/cue.binOp(0xc0088018c0, 0xd26ac0, 0xc004993780, 0xc004940001, 0xd30b60, 0xc09b608460, 0xd30b60, 0xc004945260, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/binop.go:123 +0x484 fp=0xc0089f38a8 sp=0xc0089f3788 pc=0x72baa4
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004993780, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:467 +0x378 fp=0xc0089f39e8 sp=0xc0089f38a8 pc=0x753b58
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0049ac680, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f3b28 sp=0xc0089f39e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0049acb40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f3c68 sp=0xc0089f3b28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0049ad500, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f3da8 sp=0xc0089f3c68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0049ad9c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f3ee8 sp=0xc0089f3da8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0049dc380, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4028 sp=0xc0089f3ee8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0049dd140, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4168 sp=0xc0089f4028 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004a92180, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f42a8 sp=0xc0089f4168 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004a93300, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f43e8 sp=0xc0089f42a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004aca340, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4528 sp=0xc0089f43e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004acb380, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4668 sp=0xc0089f4528 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004b1e640, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f47a8 sp=0xc0089f4668 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004b1f900, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f48e8 sp=0xc0089f47a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004b52bc0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4a28 sp=0xc0089f48e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004b53c00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4b68 sp=0xc0089f4a28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004b86c40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4ca8 sp=0xc0089f4b68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004c28e80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4de8 sp=0xc0089f4ca8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004c29ec0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f4f28 sp=0xc0089f4de8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004c49040, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5068 sp=0xc0089f4f28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004c86080, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f51a8 sp=0xc0089f5068 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004c87f00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f52e8 sp=0xc0089f51a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004cf0f40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5428 sp=0xc0089f52e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004cf9180, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5568 sp=0xc0089f5428 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004d2c1c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f56a8 sp=0xc0089f5568 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004d4a540, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f57e8 sp=0xc0089f56a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004d4b580, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5928 sp=0xc0089f57e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004d72480, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5a68 sp=0xc0089f5928 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004e44d80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5ba8 sp=0xc0089f5a68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004e45dc0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5ce8 sp=0xc0089f5ba8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004e88b80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5e28 sp=0xc0089f5ce8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004e89a80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f5f68 sp=0xc0089f5e28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004ec1b80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f60a8 sp=0xc0089f5f68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004f00180, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f61e8 sp=0xc0089f60a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004f07200, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6328 sp=0xc0089f61e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004f46100, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6468 sp=0xc0089f6328 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004f47000, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f65a8 sp=0xc0089f6468 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004f47f00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f66e8 sp=0xc0089f65a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004facfc0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6828 sp=0xc0089f66e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004fadec0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6968 sp=0xc0089f6828 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004fd3040, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6aa8 sp=0xc0089f6968 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc004fdcf40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6be8 sp=0xc0089f6aa8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005007180, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6d28 sp=0xc0089f6be8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005031e00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6e68 sp=0xc0089f6d28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005062d00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f6fa8 sp=0xc0089f6e68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050631c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f70e8 sp=0xc0089f6fa8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00507b7c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7228 sp=0xc0089f70e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050886c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7368 sp=0xc0089f7228 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00509af40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f74a8 sp=0xc0089f7368 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00509be40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f75e8 sp=0xc0089f74a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050c9240, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7728 sp=0xc0089f75e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050cc140, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7868 sp=0xc0089f7728 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050cc600, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f79a8 sp=0xc0089f7868 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050cd840, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7ae8 sp=0xc0089f79a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050e2c40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7c28 sp=0xc0089f7ae8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0050e3bc0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7d68 sp=0xc0089f7c28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00510bc40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7ea8 sp=0xc0089f7d68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005112940, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f7fe8 sp=0xc0089f7ea8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005120b00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8128 sp=0xc0089f7fe8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005124900, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8268 sp=0xc0089f8128 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051258c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f83a8 sp=0xc0089f8268 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005132540, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f84e8 sp=0xc0089f83a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005133dc0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8628 sp=0xc0089f84e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005148c00, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8768 sp=0xc0089f8628 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00515ca40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f88a8 sp=0xc0089f8768 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00515d680, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f89e8 sp=0xc0089f88a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005165300, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8b28 sp=0xc0089f89e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005165d80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8c68 sp=0xc0089f8b28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005172500, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8da8 sp=0xc0089f8c68 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005182100, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f8ee8 sp=0xc0089f8da8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc005183540, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9028 sp=0xc0089f8ee8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00518c1c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9168 sp=0xc0089f9028 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00519d680, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f92a8 sp=0xc0089f9168 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc00519df80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f93e8 sp=0xc0089f92a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051a8800, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9528 sp=0xc0089f93e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051a9a40, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9668 sp=0xc0089f9528 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051b2c80, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f97a8 sp=0xc0089f9668 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051b3780, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f98e8 sp=0xc0089f97a8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051be200, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9a28 sp=0xc0089f98e8 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051bf440, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9b68 sp=0xc0089f9a28 pc=0x75385f
cuelang.org/go/cue.(*binaryExpr).evalPartial(0xc0051cd1c0, 0xc0088018c0, 0x0, 0x0)
	/Users/mpvl/dev/release/cue/cue/eval.go:453 +0x7f fp=0xc0089f9ca8 sp=0xc0089f9b68 pc=0x75385f
```

It seems there's an infinity loop in the code.

The cue version I'm using

```
cue version 0.0.14 linux/amd64
```

I put the yaml files at https://github.com/adieu/cue-ansible so people could have a try with the data.

**Comment by cueckoo** [comment]
_Original reply by @jlongtine in https://github.com/cuelang/cue/issues/184#issuecomment-557605855_

@adieu How do you import these? What `cue` command do you run?

**Comment by cueckoo** [comment]
_Original reply by @adieu in https://github.com/cuelang/cue/issues/184#issuecomment-557612016_

```
cue import ./... -p ansible -l '"module": "\(module)"' -f
```

The first error I got is:
```
undefined field "module":
    <path flag>:1:14
```

Fix that by compile the master branch.

Then I got #185. Fixed that in #186.

Now I can import all the yaml files without error.

**Comment by cueckoo** [comment]
_Original reply by @adieu in https://github.com/cuelang/cue/issues/184#issuecomment-557889901_

I dug deeper for this OOM issue. I found it occurs when I put all the module schema under one root object. It went away if I import with ` cue import ./... -p ansible -l '"\(module)"' -f`.

After doing some profiling, I traced to `binOp(ctx, x, x.op, left, right)` in `(*binaryExpr).evalPartial` which will walk though the whole object and `ctx.copy` will be called which will consume quite a lot of memory.

I'm still reading the code and trying to come up a fix. Hoping this update would provide more context to this issue.
